File: python_open/matOp.py
import cv2
import numpy as np

def main():
    # 3채널 이미지로 초기화 (어두운 빨간색)
    img1 = np.full((400, 640, 3), (10, 0, 0), dtype=np.uint8)
    # 단일 채널 이미지로 초기화 (검정색)
    img2 = np.full((400, 640), 0, dtype=np.uint8)
    # 3채널 이미지로 초기화 (파란색)
    img3 = np.full((400, 640, 3), (255, 0, 0), dtype=np.uint8)
    # 3채널 이미지로 초기화 (초록색)
    img4 = np.full((400, 640, 3), (0, 255, 0), dtype=np.uint8)
    # 단일 채널 이미지로 초기화 (검정색)
    img5 = np.full((400, 640), 0, dtype=np.uint8)
    # 단일 채널 이미지로 초기화 (흰색)
    img6 = np.full((400, 640), 255, dtype=np.uint8)

    # 각 이미지를 출력
    cv2.imshow("img1", img1)
    cv2.imshow("img2", img2)
    cv2.imshow("img3", img3)
    cv2.imshow("img4", img4)
    cv2.imshow("img5", img5)
    cv2.imshow("img6", img6)

    cv2.waitKey(0)  # 키 입력을 기다림
    cv2.destroyAllWindows()  # 모든 창 닫기

if __name__ == "__main__":
    main()

File: cppOpen/matOp.cpp
#include "opencv2/opencv.hpp"
#include <iostream>

using namespace cv;
using namespace std;

int main() 
{
    Mat img1 = Mat(400, 640, CV_8UC3, Scalar(10, 0, 0));  // 3채널 이미지로 초기화
    Mat img2 = Mat(400, 640, CV_8UC1, Scalar(0));          // 단일 채널 이미지로 초기화
    Mat img3 = Mat(400, 640, CV_8UC3, Scalar(255, 0, 0));  // 3채널 이미지로 초기화 (파란색)
    Mat img4 = Mat(Size(640, 400), CV_8UC3, Scalar(0, 255, 0));  // 3채널 이미지로 초기화 (초록색)

    Mat img5 = Mat(400, 640, CV_8UC1, Scalar(0));          // 단일 채널 이미지로 초기화 (검정색)
    Mat img6 = Mat(400, 640, CV_8UC1, Scalar(255));        // 단일 채널 이미지로 초기화 (흰색)

    imshow("img1", img1);
    imshow("img2", img2);
    imshow("img3", img3);
    imshow("img4", img4);
    imshow("img5", img5);
    imshow("img6", img6);
    waitKey(0);
    return 0;
}
